File: android_client/app/src/main/java/com/zigzag/client_app/model/ScreenConfig.java
package com.zigzag.client_app.model;

public class ScreenConfig {
    private String key;
    private int width;

    public ScreenConfig(String key, int width) {
        this.key = key;
        this.width = width;
    }

    public String getKey() {
        return key;
    }

    public int getWidth() {
        return width;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }

        if (!(o instanceof ScreenConfig)) {
            return false;
        }

        ScreenConfig that = (ScreenConfig) o;

        if (width != that.width) {
            return false;
        }

        if (!key.equals(that.key)) {
            return false;
        }

        return true;
    }

    @Override
    public int hashCode() {
        int result = key.hashCode();
        result = 31 * result + width;
        return result;
    }
}

File: ios_client/ios_client/models/ScreenConfig.swift
/**
 * ScreenConfig.swift
 * Created by Liviu Coman on 3/24/15.
 * Copyright (c) 2015 LiviuComan. All rights reserved.
 * based on design by HoriaComan
 */

import UIKit

/**
 * Class Screen config that holds info about the
 * screen display method of an image
 */
class ScreenConfig: NSObject {
   /**
    * private variables
    */
    private var key: String! = nil
    private var width: Int! = nil
    
    /**
     * custom constructor
     */
    init(key: String, width: Int){
        super.init()
        self.key = key
        self.width = width
    }
    
    /**
     * getter functions
     */
    func getKey() -> String {
        return self.key
    }
    
    func getWidth() -> Int {
        return self.width
    }
    
    /**
     * function that acts as Equal
     */
    func equals(var o: AnyObject) -> Bool{
        if (self == o as ScreenConfig){
            return true
        }
        
        if (!(o is ScreenConfig)){
            return false;
        }
        
        if (self.width != (o as ScreenConfig).width){
            return false
        }
        
        if (self.key != (o as ScreenConfig).key){
            return false
        }
        
        return true
    }
    
    /**
     * form and get the hash key
     */
    func hashCode() -> Int {
        var result = key.hash
        return 31 * result + width
    }
}
